test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -3720,6 +3721,46 @@ public async Task TaskHubName_DefaultNameNonProductionSlot_ThrowsException()
             }
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task CustomISerializerSettingsFactory()
+        {
+            string[] orchestratorFunctionNames =
+            {
+                nameof(this.CustomISerializerSettingsFactory),
+            };
+
+            using (var host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.ReplaySafeLogger_LogsOnlyOnce),
+                true,
+                serializerSettings: new CustomEnumSettings()))
+            {
+                await host.StartAsync();
+
+                var inputWithEnum = new ComplexType
+                {
+                    A = -42,
+                    B = new List<DateTime> { DateTime.UtcNow, DateTime.UtcNow.AddYears(1) },
+                    C = ComplexType.CustomEnum.Value2,
+                    D = new ComplexType.ComplexInnerType
+                    {
+                        E = Guid.NewGuid().ToString(),
+                        F = TimeSpan.FromHours(1.5),
+                    },
+                };
+
+                var client = await host.StartOrchestratorAsync(orchestratorFunctionNames[0], inputWithEnum, this.output);
+                await client.WaitForCompletionAsync(this.output);
+                var status = client.GetStatusAsync();
+
+                Assert.NotNull(status);
+
+                var expectedResult = "Value2";
+                Assert.Contains(expectedResult, status.Result.Output.ToString());

```

Comment:
Just to make sure this is working the way we expect, I would suggest we run two orchestrations: one without your customizations and one with your customizations. That way we won't get a false positive if we accidentally change the default behavior for enum serialization in the future (which we don't want to do since that is a breaking change).